Pirat/Services/Resource/ResourceUpdateService.cs

```diff
@@ -156,19 +157,135 @@ public Task ChangeConsumableAmount(string token, int consumableId, int newAmount
             return ChangeConsumableAmount(token, consumableId, newAmount, null);
         }
 
-        public Task ChangeConsumableAmount(string token, int consumableId, int newAmount, string reason)
+        public async Task ChangeConsumableAmount(string token, int consumableId, int newAmount, string reason)
         {
-            throw new NotImplementedException();
+            // Get consumable from database
+            var query = from o in _context.offer
+                join c in _context.consumable on o.id equals c.offer_id
+                where token.Equals(o.token)
```

Comment:
Are you sure this works? Using equals in a query has given me always a database exception. The solution for me was to use == since it will be a sql query.

Reply:
Well, the tests say yes...?